### ISSUE TITLE
Feature Request: Submitting URL or IP should allow users to pick country/region the proxy will egress from.

### ISSUE THREAD

**Issue by C-Oliver** [CybercentreCanada/assemblyline#58]
**Is your feature request related to a problem? Please describe.**
When submitting a domain or IP, users should be allowed to pick what country region to egress out of, maybe a general setting in the UI configurable per user would be a way to do this.

**Describe the solution you'd like**
In some cases we want to egress out of a specific country when submitting a domain or IP in Assemblyline, the only way to do this at the moment is to update the backend proxy settings, we should be able to do this through the UI.

**Describe alternatives you've considered**
Update values.yaml manually.

**Additional context**
N/A

**Comment by cccs-rs** [comment]
So in this case, you'd like to have some kind of menu that appears when a URL is inputted that allows you to specific the proxy to egress from?

Quick sketch:
![image](https://github.com/CybercentreCanada/assemblyline/assets/62077998/d76f65e7-c088-47a5-a9d2-27b5a6e337f3)

In the case, the different egress points would be a system admin configuration and the users would see the "pretty" names in the dropdown and use that in their submissions as a parameter.

**Comment by ociappara** [comment]
This would be ideal yes!

**Comment by cccs-rs** [comment]
Current render of providing egress options in to the UI (only when a URL is given):
![image](https://github.com/CybercentreCanada/assemblyline/assets/62077998/b87f0c0c-7c8d-4dec-8c37-ad4a30a89bf5)

Example config:
```yaml
config:
  ...
  ui:
    url_egress_proxies:
      CANADA:
        https: a
      GeRMANY:
        http: b
      united_states:
        http: c
```

So far the frontend will tidy up the proxy key names but the true value that gets sent back to the Submit API is the original.

An idea that came to mind, do you foresee a need to block users from submitting through certain proxies?

**Comment by C-Oliver** [comment]
Maybe restriction by Groups? Also, in our case, the URL for the proxy is the same, the port is what dictates what country to egress out from, does the current implementation take into consideration the port?

**Comment by cccs-rs** [comment]
The current implementation allows you to specify the proxy dict that gets passed to the Requests library. So it would be possible to have a configuration like:

```yaml
config:
  ...
  ui:
    url_egress_proxies:
      CANADA:
        https: "https://my-proxy:1000"
      GeRMANY:
        http: "https://my-proxy:2000"
      united_states:
        http: "https://my-proxy:3000"
```

**Comment by C-Oliver** [comment]
> The current implementation allows you to specify the proxy dict that gets passed to the Requests library. So it would be possible to have a configuration like:
> 
> ```yaml
> config:
>   ...
>   ui:
>     url_egress_proxies:
>       CANADA:
>         https: "https://my-proxy:1000"
>       GeRMANY:
>         http: "https://my-proxy:2000"
>       united_states:
>         http: "https://my-proxy:3000"
> ```

That's great thanks @cccs-rs !

**Comment by cccs-rs** [comment]
This should be patched in the [4.5.0.20](https://github.com/CybercentreCanada/assemblyline/releases/tag/v4.5.0.stable20) release of Assemblyline.

You can create an internal service on the system that's configured for URL fetching through a proxy via the File Sources configuration. Feel free to re-open the issue if necessary.
